File: 118 Pascal's Triangle/118-2.js
var generate = function(numRows){
    //宣告陣列長
    var x = new Array();
    for( var i = 0; i < numRows; i++){
        //宣告各陣列寬
        var x[i] = new Array();
        x[i][i]=1;
        x[i][0]=1;
        for( var j = 0; j <= i; j++ ){
            //取上層相加
            x[i][j] = x[i-1][j] + x[i-1][j-1];
        }
    }
    return x;
}
//Min:116ms

File: 118 Pascal's Triangle/118-2.py
class Solution(object):
    def generate(self, numRows):
        if numRows == 0:
            return []
        x=[[1]]
        if numRows == 1:
            return x
        for I in range(1, numRows):
            x.append([])
            x[i].append(1)
            for j in range(1, i):
                x[i].append(x[i-1][j] + x[i-1][j])
            x[i].append(1)
        return x
#Min:36ms
